### ISSUE TITLE
react-hooks/rules-of-hooks misses a Hook after an early return in a component

### ISSUE THREAD

**Issue by fireairforce** [utooland/utoo-lint#1221]
## Summary

With `@utoo/lint@0.4.1`, the native CLI accepts `react-hooks/rules-of-hooks: error` but does not report a Hook called after an early return in a component. ESLint 8.57.1 with eslint-plugin-react-hooks 4.6.2 reports the conditional Hook call.

## Minimal reproduction

```sh
npm init -y
npm install --save-dev @utoo/lint@0.4.1 eslint@8.57.1 eslint-plugin-react-hooks@4.6.2 @typescript-eslint/parser@6.21.0
```

`example.tsx`:

```tsx
import { useEffect } from 'react';
export default function Example({ ready }) {
  if (!ready) { return null; }
  useEffect(() => {}, []);
  return null;
}
```

`utlint.config.json`:

```json
{"rules":{"react-hooks/rules-of-hooks":"error"}}
```

`.eslintrc.cjs`:

```js
module.exports = {
  root: true,
  parser: '@typescript-eslint/parser',
  parserOptions: { ecmaVersion: 2022, sourceType: 'module', ecmaFeatures: { jsx: true } },
  plugins: ['react-hooks'],
  rules: { 'react-hooks/rules-of-hooks': 'error' },
};
```

```sh
./node_modules/.bin/utlint --config utlint.config.json --format=json example.tsx
./node_modules/.bin/eslint --no-eslintrc --config .eslintrc.cjs --format=json example.tsx
```

## Actual and expected

- utlint: one file checked, empty `diagnostics`, exit 0.
- ESLint: one `react-hooks/rules-of-hooks` error at line 4, column 3, exit 1.
- Expected: report the conditional Hook call because the early return changes the number of Hooks called between renders.

Verified on macOS arm64 with Node.js 22.23.1. This is a standalone example with no custom plugins or project-specific configuration.
